src/core/space_interaction_API.jl

```diff
@@ -140,11 +165,15 @@ end
 
 Remove an agent from the model.
 """
-function kill_agent!(a::AbstractAgent, model::ABM)
-    delete!(model.agents, a.id)
+function kill_agent!(a::AbstractAgent, model::ABM{S,A,Dict{Int,A}}) where {S,A}
+    remove_agent_from_model!(a, model)
     remove_agent_from_space!(a, model)
 end
 
+function kill_agent!(a::AbstractAgent, model::ABM{S,A,<:AbstractVector{A}}) where {S,A}
+    nokill()
+end
+
```

Comment:
Isn't this unecessary? Instead of having two call signatures you can just make the `remove_agent_from_model!(a, model)` function throw this error for SizedArray.